### PR TITLE
Initial Db setup in docker and local with one table

### PR DESCRIPTION
**Pull request by kryswisnaskas**
**Description of change**
This PR configures Postgres database as well as a model and a migration for Users table. It also creates a function that accesses or creates a user.

**How to test**
**In the container:**
- update `.env` file with the environment variables and the password from above
- `yarn docker:deps`
- `yarn docker:db-migrate`
- `yarn docker:test`
- the tests
Bonus:
- `yarn docker:db-migrate:undo`
- `yarn docker:test`
- note that the tests fail now

**Locally:**
- install Postgres and set a password for user "postgres"
- using your favorite client, e.g. PGAdmin or psql create the database `ttasmarthub` (`create database ttasmarthub`)
- update .env file with the environment variables and the password from above
- `yarn deps`
- `yarn db-migrate`
- `yarn test`
- the tests should pass

**Issue(s)**
https://github.com/HHS/Head-Start-TTADP/issues/16

**Checklist**

- [x] Meets issue criteria
- [x] Code tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [x] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

File: src/services/accessValidation.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * Finds or creates a user in the database.
+ *
+ * //potentially use also or instead of the user id provided by HSES
+ * @param {Object} userData - user information containing email address
+ * @returns {Promise<any>} - returns a promise
+ */
+export default function findOrCreateUser(data) {
+  return sequelize.transaction((transaction) => User.findOrCreate({
+    where: {
+      email: data.email,
```
Comment (jasalisbury) [inline src/services/accessValidation.js:16]: Should this be something like:
```suggestion
      ...data,
```
so the data object can have more then just the user's email address?

**Review by jasalisbury — COMMENTED** [review]
Looks good! Have some pretty minor questions

File: package.json [hunk 2]
```diff
@@ -19,13 +19,18 @@
     "lint:ci": "eslint -f eslint-formatter-multiple src",
     "lint:all": "yarn lint:ci && yarn --cwd frontend lint:ci",
     "clean": "rm -rf coverage reports frontend/coverage frontend/reports frontend/build",
+    "db-migrate": "node_modules/.bin/sequelize db:migrate",
+    "db-migrate-undo": "node_modules/.bin/sequelize db:migrate:undo",
```
[the remark this answers is not in the document]
Comment (kryswisnaskas) [reply package.json:23]: No problem.

File: .sequelizerc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const path = require('path');
+
+module.exports = {
+  'config': path.resolve('config', 'config.js'),
+  'models-path': path.resolve('src', 'models'),
+  'seeders-path': path.resolve('src','models', 'seeders'),
+  'migrations-path': path.resolve('src', 'models', 'migrations')
```
[the remark this answers is not in the document]
Comment (kryswisnaskas) [reply .sequelizerc:7]: I can change it, if this makes more sense/it's easier to look at.

File: src/models/migrations/20200827192413-create-user.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+module.exports = {
+  up: (queryInterface, Sequelize) => queryInterface.createTable('Users', {
+    userId: {
```
[the remark this answers is not in the document]
Comment (kryswisnaskas) [reply src/models/migrations/20200827192413-create-user.js:3]: There are 2 schools of thought on this. `id `is `sequelize's `default. Having a table name in the identity column has its advantages. It all comes down to preference. I am not attached to either way of handling the ids. If we like the plain `id `without the table name in it, can adjust.

[the remark this answers is not in the document]
Comment (jasalisbury) [reply src/models/migrations/20200827192413-create-user.js:3]: I can go either way as well. My rational is when I see `${tableName}Id` I think foreign key, but no big deal either way.

[the remark this answers is not in the document]
Comment (SarahJaine) [reply src/models/migrations/20200827192413-create-user.js:3]: FWIW I'm also used to Table.id for the primary keys and tableId for foreign keys.

File: src/models/migrations/20200827192413-create-user.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      type: Sequelize.INTEGER,
+      allowNull: false,
+      primaryKey: true,
+      autoIncrement: true,
+    },
+    name: {
```
Comment (SarahJaine) [inline src/models/migrations/20200827192413-create-user.js:9]: I'm possibly overthinking this for a sprint 0 ticket but we might want `first` `last` as separate cols. The UI might want to do welcome and user messages with just the first name.  Trying to split out a combined field on the frontend is usually error prone.

File: src/models/migrations/20200827192413-create-user.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (kryswisnaskas) [reply src/models/migrations/20200827192413-create-user.js]: Changed.

File: src/models/migrations/20200827192413-create-user.js (hunk 5 above)
Comment (kryswisnaskas) [reply src/models/migrations/20200827192413-create-user.js]: Yes, `name `was meant more of a placeholder, but if we know how the data is going to be structured, will change it. I'll investigate.

**Review by jasalisbury — APPROVED** [review]
LGTM

**Pull request merged**